src/Corax/Queries/SortingMatch.Comparers.cs

```diff
@@ -20,7 +15,158 @@ internal static int CompareAscending(ReadOnlySpan<byte> x, ReadOnlySpan<byte> y)
             }
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            internal static int CompareAscending<T>(T x, T y)
+            private static bool IsSpace(byte c)
+            {
+                return c == ' ';
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private static bool IsDigit(byte c)
+            {
+                return c >= '0' && c <= '9';
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private static byte ToUpper(byte c)
+            {
+                if (c >= 'a' && c <= 'z')
+                    return (byte)(c - ('a' - 'A'));
+
+                return c; 
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static int CompareAlphanumericAscending(ReadOnlySpan<byte> a, ReadOnlySpan<byte> b, bool foldCase = false)
```

Comment:
This is valid only to ASCII buffers

Reply:
@redknightlois

Reply:
Yes, we need to check the buffer for ASCIIness and go back to standard comparer. Unless you know how to do that with Unicode (UTF-8).

Reply:
Alpha numeric is rare enough that we don't need to specialize here. Can always use the usual comparere.